### PR TITLE
Add Vega Keyboard

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the Vega keyboard.  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for Vega keyboard

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/vega/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_names {
+    _BASE,
+    _FN1,
+	_FN2,
+	_FN3
```
Comment (drashna) [inline keyboards/ai03/vega/keymaps/via/keymap.c:28]: tabs vs spaces
```suggestion
    _FN2,
    _FN3
```

File: keyboards/ai03/vega/vega.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include "vega.h"
```
Comment (drashna) [inline keyboards/ai03/vega/vega.c:1]: Could you include a license header in this file, please?

File: keyboards/ai03/vega/vega.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/ai03/vega/vega.h:1]: Could you include a license header in this file, please?

File: keyboards/ai03/vega/config.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xA103
+#define PRODUCT_ID      0x0015
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    ai03 Design Studio
+#define PRODUCT         Vega
+#define DESCRIPTION     65% keyboard
```
Comment (fauxpark) [inline keyboards/ai03/vega/config.h:26]:
```suggestion
```

File: keyboards/ai03/vega/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+A typical 65% keyboard  
+
+* Keyboard Maintainer: [ai03](https://github.com/ai03-2725)
+* Hardware Supported: Vega PCBs, both solderable and hotswap
+* Hardware Availability: Check the Vega page on [ai03's website](https://ai03.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ai03/vega:via
```
Comment (fauxpark) [inline keyboards/ai03/vega/readme.md:13]:
```suggestion
    make ai03/vega:default
```

File: keyboards/ai03/vega/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Vega
+
+![Vega](https://i.imgur.com/7ZcCqwP.png)
+
+A typical 65% keyboard
```
Comment (fauxpark) [inline keyboards/ai03/vega/readme.md:5]:
```suggestion
A typical 65% keyboard
```

File: keyboards/ai03/vega/rules.mk [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no      # Custom matrix file
+BACKLIGHT_ENABLE = no   # Backlight 
+RGBLIGHT_ENABLE = no    # RGB Underglow
```
Comment (fauxpark) [inline keyboards/ai03/vega/rules.mk:16]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ai03/vega/vega.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_default( \
```
Comment (noroadsleft) [inline keyboards/ai03/vega/vega.h:21]:
```suggestion
#define LAYOUT_all( \
```

File: keyboards/ai03/vega/info.json [hunk 9]
```diff
@@ -0,0 +1,379 @@
+{
+    "keyboard_name": "Vega",
+    "url": "ai03.com",
+    "maintainer": "ai03",
+    "width": 16,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/ai03/vega/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/ai03/vega/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/default/keymap.c:31]:
```suggestion
	[_BASE] = LAYOUT_all(
```

File: keyboards/ai03/vega/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_HOME, 
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS, KC_END,  
+		MO(1),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP, 
+		KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_PGDN, 
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_SPC,           KC_SPC,           KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT 
+    ),
+
+    [_FN] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/default/keymap.c:39]:
```suggestion
    [_FN] = LAYOUT_all(
```

File: keyboards/ai03/vega/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+enum layer_names {
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/via/keymap.c:33]:
```suggestion
	[_BASE] = LAYOUT_all(
```

File: keyboards/ai03/vega/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_HOME, 
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS, KC_END,  
+		MO(1),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP, 
+		KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_PGDN, 
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_SPC,           KC_SPC,           KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT 
+	),
+
+	[_FN1] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/via/keymap.c:41]:
```suggestion
	[_FN1] = LAYOUT_all(
```

File: keyboards/ai03/vega/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+		RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, _______, 
+		_______, _______, _______, KC_PGUP, _______, _______, _______, _______, KC_UP,   _______, _______, _______, _______,          _______, _______, 
+		_______, _______, KC_HOME, KC_PGDN, KC_END,  _______, _______, KC_LEFT, KC_DOWN, KC_RGHT, _______, _______,          _______,          _______, 
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, KC_PGUP, _______, 
+		_______, _______, _______,          _______,          _______,          _______,          _______, _______,          KC_HOME, KC_PGDN, KC_END 
+	),
+
+	[_FN2] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/via/keymap.c:49]:
```suggestion
	[_FN2] = LAYOUT_all(
```

File: keyboards/ai03/vega/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, 
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______, 
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, 
+		_______, _______, _______,          _______,          _______,          _______,          _______, _______,          _______, _______, _______  
+	),
+
+	[_FN3] = LAYOUT_default(
```
Comment (noroadsleft) [inline keyboards/ai03/vega/keymaps/via/keymap.c:57]:
```suggestion
	[_FN3] = LAYOUT_all(
```

File: keyboards/ai03/vega/readme.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
+# Vega
+
+![Vega](https://i.imgur.com/7ZcCqwP.png)
```
Comment (noroadsleft) [inline keyboards/ai03/vega/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![Vega](https://i.imgur.com/7ZcCqwPl.png)
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**